base/hashing2.jl

```diff
@@ -13,16 +13,19 @@ function hash_integer(n::Integer, h::UInt)
     return h
 end
 
-function hash_integer(n::BigInt, h::UInt)
-    s = n.size
-    s == 0 && return hash_integer(0, h)
-    p = convert(Ptr{UInt}, n.d)
-    b = unsafe_load(p)
-    h ⊻= hash_uint(ifelse(s < 0, -b, b) ⊻ h)
-    for k = 2:abs(s)
-        h ⊻= hash_uint(unsafe_load(p, k) ⊻ h)
+if GMP.Limb === UInt
```

Comment:
Seems like it would be worth adding a comment noting what happens when this isn't true. Am I reading it correctly that when `GMP.Limb !== UInt` this falls back to the above `hash_integer(n::Integer, h::UInt)` method?

Reply:
Yes this then falls back to the slower `hash_integer(::Integer, ::UInt)`. Ok, will add a comment. I'm not even certain this happens in practive, but I couldn't prove that it can't, so it seemed safer to state clearly the assumptions.

Reply:
Before pushing again and running CI for only comments change, let's check first if the comment is fine:
```
# this condition is true most (all?) of the time, and in this case we can define
# an optimized version of the above hash_integer(::Integer, ::UInt) method for BigInt
if GMP.Limb === UInt
...
```
```

Reply:
Sure, looks good. You could also just push a separate `[ci skip]` commit and then we can squash on merge.

Reply:
Ok will do, for some reason I stayed with the idea that `[ci skip]` didn't work anymore, since like 2 years ago or so!